[Drop-In UI] add option to specify custom location puck with drop-in

Zayankovsky opened the pull request:
<!--
---------- CHECKLIST ----------
1. Add related labels (`bug`, `feature`, `new API(s)`, `SEMVER-MAJOR`, `needs-backporting`, etc.).
1. Adda a changelog entry under `Unreleased` tag or a `skip changelog` label if not applicable.
1. Update progress status on the project board.
1. Request a review from the team, if not a draft.
1. Add targeted milestone, when applicable.
1. Create ticket tracking addition of public documentation pages entry, when applicable.
-->

tomaszrybakiewicz reviewed (approved):
LGTM
@Zayankovsky Can you add description to this PR?

abhishek1508 commented on CHANGELOG.md:
```diff
@@ -7,6 +7,7 @@ Mapbox welcomes participation and contributions from everyone.
 - Moved `MapboxAudioGuidance` and `MapboxAudioGuidanceState` into public api. [#6336](https://github.com/mapbox/mapbox-navigation-android/pull/6336)
 - Introduced `ViewOptionsCustomization.showCameraDebugInfo` to allow end users to enable camera debug info. [#6356](https://github.com/mapbox/mapbox-navigation-android/pull/6356)
 - Added `ComponentInstaller` for the `LocationComponent` that offers simplified integration of map LocationPuck. [#6206](https://github.com/mapbox/mapbox-navigation-android/pull/6206)
+- Added `ViewStyleCustomization.locationPuck` that allows to define a custom location puck for `MapView`. [#6365](https://github.com/mapbox/mapbox-navigation-android/pull/6365)
```
```suggestion
- Added `ViewStyleCustomization.locationPuck` that allows to define a custom location puck using `NavigationView`. [#6365](https://github.com/mapbox/mapbox-navigation-android/pull/6365)
- Replaced `ViewStyleCustomization.defaultMarkerAnnotationOptions` with `ViewStyleCustomization.defaultDestinationMarkerAnnotationOptions`
```

abhishek1508 commented on libnavigation-core/api/current.txt:
```diff
@@ -676,11 +676,13 @@ package com.mapbox.navigation.core.routealternatives {
   }
 
   public final class AlternativeRouteMetadata {
+    method public int getAlternativeId();
```
Did we miss these docs in some other PR?

Zayankovsky replied:
Yes, @VysotskiVadim forgot to update these files in #6329.

abhishek1508 reviewed (commented):
@Zayankovsky

tomaszrybakiewicz commented:
@abhishek1508 
That stutter is definitely on the Maps side. Puck position is animated like this:

<img src="https://user-images.githubusercontent.com/2678039/176534659-c244d613-0d4d-40a9-a915-87eb95d1d01b.png" />

I can only assume that when the map style changes mid-animation, the animated style layer is destroyed along with its animator. The remaining animation frames are "lost". 
When the new style layer is installed, the new animation starts on next `LocationComponentPluginImpl.updateCurrentPosition(...)` call (which is every second), hence the stutter.

abhishek1508 reviewed (approved):
Thanks for addressing the feedback!

Zayankovsky commented:
> @Zayankovsky I am seeing a stutter when changing the puck at runtime while in active navigation. 

It should be better now, looks like it was caused by repeatedly setting the same location provider, so I added a check to avoid this. Also I removed a bunch of code from `LocationPuckComponent`, we don't need to wait for location updates or for style to load before updating location plugin.

dzinad commented on libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/ui/LocationPuckComponent.kt:
```diff
@@ -25,49 +17,12 @@ class LocationPuckComponent(
 
-        coroutineScope.launch {
-            locationProvider.waitForFirstLocation()
-            map.getStyle {
```
Why is it not needed anymore? And waiting for the first location.

Zayankovsky replied:
Waiting for the first location was added in #5555. I think it was an issue with `getStyle`, which is now also removed. We don't do any of that in 1TAP and never had any issues with location puck.
